components/brave_wallet/browser/brave_wallet_controller.cc

```diff
@@ -0,0 +1,216 @@
+      !profile->GetPrefs()->HasPrefPath(kBraveWalletEncryptedSeed)) {
+    return false;
+  }
+  if (!base::Base64Decode(
+          profile->GetPrefs()->GetString(kBraveWalletAES256GCMSivNonce),
+          nonce)) {
+    return false;
+  }
+  if (!base::Base64Decode(
+          profile->GetPrefs()->GetString(kBraveWalletEncryptedSeed),
+          cipher_seed)) {
+    return false;
+  }
+  return true;
+}
+
+// static
+bool BraveWalletController::OpenSeed(const std::string& cipher_seed,
+    const std::string& key, const std::string& nonce,
+    std::string* seed) {
+  crypto::Aead aes_256_gcm_siv(crypto::Aead::AES_256_GCM_SIV);
+  aes_256_gcm_siv.Init(&key);
+  return aes_256_gcm_siv.Open(cipher_seed, nonce, base::StringPiece(""), seed);
+}
+
+// Generate a new random nonce
+// static
+std::string BraveWalletController::GetRandomNonce() {
+  // crypto::RandBytes is fail safe.
+  uint8_t nonceBytes[kNonceByteLength];
+  crypto::RandBytes(nonceBytes, kNonceByteLength);
+  return std::string(
+      reinterpret_cast<char*>(nonceBytes), kNonceByteLength);
+}
+
+// static
+std::string BraveWalletController::GetRandomSeed() {
+  // crypto::RandBytes is fail safe.
+  uint8_t random_seed_bytes[kSeedByteLength];
+  crypto::RandBytes(random_seed_bytes, kSeedByteLength);
+  return std::string(
+      reinterpret_cast<char*>(random_seed_bytes), kSeedByteLength);
+}
+
+// static
+bool BraveWalletController::SealSeed(const std::string& seed,
+    const std::string& key, const std::string& nonce,
+    std::string* cipher_seed) {
+  crypto::Aead aes_256_gcm_siv(crypto::Aead::AES_256_GCM_SIV);
+  aes_256_gcm_siv.Init(&key);
+  return aes_256_gcm_siv.Seal(base::StringPiece(seed.begin(), seed.end()),
+      nonce, base::StringPiece(""), cipher_seed);
+}
+
+// Store the seed in preferences, binary pref strings need to be
+// base64 encoded.  Base64 encoding is fail safe.
+// static
+void BraveWalletController::SaveToPrefs(
+    Profile* profile,
+    const std::string& cipher_seed,
+    const std::string& nonce) {
+  // Store the seed in preferences, binary pref strings need to be
+  // base64 encoded.  Base64 encoding is fail safe.
+  std::string base64_nonce;
+  std::string base64_cipher_seed;
+  base::Base64Encode(nonce, &base64_nonce);
+  base::Base64Encode(base::StringPiece(cipher_seed.begin(),
+                                      cipher_seed.end()),
+      &base64_cipher_seed);
+  profile->GetPrefs()->SetString(kBraveWalletAES256GCMSivNonce, base64_nonce);
+  profile->GetPrefs()->SetString(kBraveWalletEncryptedSeed,
+      base64_cipher_seed);
+}
+
+void BraveWalletController::ResetCryptoWallets(
+    ResetCryptoWalletsCallback callback) {
+  Profile* profile = Profile::FromBrowserContext(context_);
+  const base::FilePath wallet_data_path = profile->GetPath()
+      .AppendASCII("Local Extension Settings")
+      .AppendASCII(ethereum_remote_client_extension_id)
+      .AppendASCII("000003.LOG");
```

Comment:
It appears in rare cases this filename can differ, better to just remove the directory

Reply:
now just removing dir